tests/integration/Tribe/Events/Aggregator/Record/Queue_CleanerTest.php

```diff
@@ -0,0 +1,252 @@
+		$failed_post_arr = [
+			'post_type'   => Records::$post_type,
+			'post_status' => Records::$status->failed
+		];
+		$failed = $this->factory()->post->create_many( 4, $failed_post_arr );
+
+		$sut = $this->make_instance();
+		$deleted = $sut->remove_duplicate_pending_records_for( $record );
+
+		$this->assertCount( 3, $deleted );
+		$this->assertCount( 1, array_diff( array_merge( $more_pending, [ $record->post->ID ] ), $deleted ) );
+	}
+
+	/**
+	 * maybe_fail_stalled_record will not fail record that has not been stalling for long
+	 */
+	public function test_maybe_fail_stalled_record_will_not_fail_record_that_has_not_been_stalling_for_long() {
+		$record = $this->getMockBuilder( Record::class )->disableOriginalConstructor()->getMock();
+		$import_id = uniqid();
+		$record->meta = [ 'import_id' => $import_id ];
+		$post_array = [ 'post_type' => Records::$post_type, 'post_status' => Records::$status->pending ];
+		$record->post = $this->factory()->post->create_and_get( $post_array );
+		add_post_meta( $record->post->ID, '_tribe_aggregator_import_id', $import_id );
+
+		$sut = $this->make_instance();
+		$sut->set_stall_limit( HOUR_IN_SECONDS );
+		$failed = $sut->maybe_fail_stalled_record( $record );
+
+		$this->assertFalse( $failed );
+		$this->assertEquals( Records::$status->pending, ( get_post( $record->post->ID ) )->post_status );
+	}
+
+	/**
+	 * maybe_fail_stalled_record will fail record that has been stalling for too long
+	 */
+	public function test_maybe_fail_stalled_record_will_fail_record_that_has_been_stalling_for_too_long() {
+		$record = $this->getMockBuilder( Record::class )->disableOriginalConstructor()->getMock();
+		$import_id = uniqid();
+		$record->meta = [ 'import_id' => $import_id ];
+		$status = Records::$status->pending;
+		$hours_ago = date( 'Y-m-d H:i:s', time() - 4 * HOUR_IN_SECONDS );
+		$post_array = [ 'post_type' => Records::$post_type, 'post_status' => $status, 'post_date' => $hours_ago ];
+		$record->post = $this->factory()->post->create_and_get( $post_array );
+		add_post_meta( $record->post->ID, '_tribe_aggregator_import_id', $import_id );
+		add_post_meta( $record->post->ID, '_tribe_aggregator_queue', 'fetch' );
+		\Tribe__Post_Transient::instance()->set( $record->post->ID, '_tribe_aggregator_queue', HOUR_IN_SECONDS );
+
+		$sut = $this->make_instance();
+		$sut->set_stall_limit( HOUR_IN_SECONDS );
+		$failed = $sut->maybe_fail_stalled_record( $record );
+
+		$this->assertTrue( $failed );
+		$this->assertEquals( Records::$status->failed, ( get_post( $record->post->ID ) )->post_status );
+		$this->assertEmpty( get_post_meta( $record->post->ID, '_tribe_aggregator_queue', true ) );
+		$this->assertEmpty( get_post_meta( $record->post->ID, '_transient_tribe_aggregator_queue', true ) );
+		$this->assertEmpty( get_post_meta( $record->post->ID, '_transient_timeout_tribe_aggregator_queue', true ) );
+	}
+
+	    /**
+	         * maybe_fail_stalled_record will empty queue meta for failed records
```

Comment:
This probably doesn't result from the merge itself - but the indentation is all out of whack here